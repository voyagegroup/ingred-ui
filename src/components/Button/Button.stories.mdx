import { Meta, Story, ArgsTable, Canvas } from "@storybook/addon-docs/blocks";
import Button from "./Button";

<Meta
  title="Components/Inputs/Button"
  component={Button}
  argTypes={{ onClick: { action: "clicked" } }}
/>

# Button

<ArgsTable of={Button} />

## Samples

### Colors

<Canvas>
  <Story
    name="primary"
    args={{
      color: "primary",
      children: "primary",
    }}
  >
    {(args) => <Button {...args} />}
  </Story>
  <Story
    name="secondary"
    args={{
      color: "secondary",
      children: "secondary",
    }}
  >
    {(args) => <Button {...args} />}
  </Story>
  <Story
    name="danger"
    args={{
      color: "danger",
      children: "danger",
    }}
  >
    {(args) => <Button {...args} />}
  </Story>
</Canvas>

### Sizes

<Canvas>
  <Story
    name="small"
    args={{
      size: "small",
      children: "small",
    }}
  >
    {(args) => <Button {...args} />}
  </Story>
  <Story
    name="medium"
    args={{
      size: "medium",
      children: "medium",
    }}
  >
    {(args) => <Button {...args} />}
  </Story>
  <Story
    name="large"
    args={{
      size: "large",
      children: "large",
    }}
  >
    {(args) => <Button {...args} />}
  </Story>
</Canvas>

### Disabled

<Canvas>
  <Story
    name="disabled"
    args={{
      disabled: true,
      children: "disabled",
    }}
  >
    {(args) => <Button {...args} />}
  </Story>
</Canvas>

### Use "href" props

<Canvas>
  <Story
    name="Use href props"
    args={{
      href: "https://google.com",
      children: "Link to google.com",
    }}
  >
    {(args) => <Button {...args} />}
  </Story>
</Canvas>

### Use with router library like "react-router"

Here's an example code.

But We have plan to change like [Material-UI](https://material-ui.com/guides/composition/#link).

```tsx
// Get "className" prop to enable <Button /> style.
const LinkToHome = ({ className, children }) => (
  // "href" prop defines not via <Button />
  <Link className={className} href="/home">
    {children}
  </Link>
);

const ButtonWithRouterLink = ({ children }) => (
  <Button component={LinkToHome}>{children}</Button>
);
```
