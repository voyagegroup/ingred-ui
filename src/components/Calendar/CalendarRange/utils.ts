import dayjs, { Dayjs } from "dayjs";

const isStart = (startDate: Dayjs | null, month: Dayjs, day: number) =>
  startDate?.format("YYYY-MM-DD") ===
  dayjs(new Date(month.year(), month.month(), day)).format("YYYY-MM-DD");

const isEnd = (endDate: Dayjs | null, month: Dayjs, day: number) =>
  endDate?.format("YYYY-MM-DD") ===
  dayjs(new Date(month.year(), month.month(), day)).format("YYYY-MM-DD");

const isBetween = (
  startDate: Dayjs | null,
  endDate: Dayjs | null,
  month: Dayjs,
  day: number,
) =>
  (startDate &&
    endDate &&
    dayjs(new Date(month.year(), month.month(), day)).isAfter(
      startDate.format("YYYY-MM-DD"),
      "day",
    ) &&
    dayjs(new Date(month.year(), month.month(), day)).isBefore(
      endDate.format("YYYY-MM-DD"),
      "day",
    )) ??
  false;

export const getDayState = (
  startDate: Dayjs | null,
  endDate: Dayjs | null,
  month: Dayjs,
  day: number,
) => {
  if (isStart(startDate, month, day)) {
    return "start";
  }
  if (isEnd(endDate, month, day)) {
    return "end";
  }
  if (isBetween(startDate, endDate, month, day)) {
    return "between";
  }
  return "none";
};
