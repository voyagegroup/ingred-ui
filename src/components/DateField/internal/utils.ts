export const getInputWidth = (input: string) =>
  [...input.split("")].reduce((acc, char) => {
    const code = char.charCodeAt(0);
    return acc + (code >= 0x3000 && code <= 0xff60 ? 2 : 1);
  }, 0);
