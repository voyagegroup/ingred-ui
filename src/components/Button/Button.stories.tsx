import React from "react";
import { Story } from "@storybook/react/types-6-0";
import {
  Title,
  Subtitle,
  Description,
  ArgsTable,
  Stories,
} from "@storybook/addon-docs";
import { action } from "@storybook/addon-actions";
import Button, { ButtonProps } from "./Button";

export default {
  title: "Components/Inputs/Button",
  components: Button,
  args: {
    onClick: action("clicked"),
  },
  parameters: {
    docs: {
      source: { type: "code" },
      page: () => (
        <>
          <Title />
          <Subtitle />
          <ArgsTable of={Button} />
          <Stories includePrimary title="Samples" />
          <Description
            markdown={[
              '### Use with router library like "react-router"',
              "Here's an example code.  ",
              "",
              "But We have plan to change like [Material-UI](https://material-ui.com/guides/composition/#link).",
              "",
              "```tsx",
              '// Get "className" prop to enable <Button /> style.',
              "const LinkToHome = ({ className, children }) => (",
              '// "href" prop defines not via <Button />',
              '  <Link className={className} href="/home">',
              "    {children}",
              "  </Link>",
              ");",
              "",
              "const ButtonWithRouterLink = ({ children }) => (",
              "  <Button component={LinkToHome}>{children}</Button>",
              ");",
              "```",
            ].join("\n")}
          />
        </>
      ),
    },
  },
};

const Template: Story<ButtonProps> = (args) => <Button {...args} />;

export const Primary = Template.bind({});
Primary.args = {
  children: "primary",
};

export const Secondary = Template.bind({});
Secondary.args = {
  children: "secondary",
  color: "secondary",
};

export const Danger = Template.bind({});
Danger.args = {
  children: "danger",
  color: "danger",
};

export const Clear = Template.bind({});
Clear.args = {
  children: "clear",
  color: "clear",
};

export const Small = Template.bind({});
Small.args = {
  children: "small",
  size: "small",
};

export const Medium = Template.bind({});
Medium.args = {
  children: "medium",
  size: "medium",
};

export const Large = Template.bind({});
Large.args = {
  children: "large",
  size: "large",
};

export const Disabled = Template.bind({});
Disabled.args = {
  children: "disabled",
  disabled: true,
};

export const UseHrefProps = Template.bind({});
UseHrefProps.args = {
  children: "use href props",
  href: "https://www.google.com",
};
