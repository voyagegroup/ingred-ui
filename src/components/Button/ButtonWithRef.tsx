import * as React from "react";
import { ButtonProps, ButtonHeight, FontSize, BorderRadius, paddingInline, ButtonColor } from "./Button";
import { useTheme } from "../../themes";
import Spacer from "../Spacer";

const ButtonWithRef = React.forwardRef<HTMLButtonElement | HTMLAnchorElement, ButtonProps>(
  function ButtonWithRef(
    {
      component,
      children,
      color = "primary",
      inline = false,
      size = "medium",
      icon,
      ...rest
    },
    ref
  ) {
    const theme = useTheme();
    
    const getColorStyle = (color: ButtonColor) => {
      switch (color) {
        case "primary":
          return {
            normal: {
              background: theme.palette.primary.main,
              color: theme.palette.text.white,
              boxShadow: theme.shadow["3dShadowPrimary"],
              border: `1px solid ${theme.palette.primary.dark}`,
            },
            hover: {
              background: theme.palette.primary.dark,
              border: "none",
            },
            active: {
              background: theme.palette.primary.dark,
              boxShadow: theme.shadow["3dShadowActive"],
              border: "none",
            },
          };
        case "basicLight":
          return {
            normal: {
              background: theme.palette.background.default,
              color: theme.palette.black,
              boxShadow: theme.shadow["3dShadowBasic"],
              border: `1px solid ${theme.palette.divider}`,
            },
            hover: {
              background: theme.palette.gray.highlight,
              border: `1px solid ${theme.palette.divider}`,
            },
            active: {
              background: theme.palette.gray.highlight,
              boxShadow: theme.shadow["3dShadowActive"],
              border: `1px solid ${theme.palette.divider}`,
            },
          };
        case "basicDark":
          return {
            normal: {
              background: theme.palette.basicDark.main,
              color: theme.palette.black,
              boxShadow: theme.shadow["3dShadowBasic"],
              border: `1px solid ${theme.palette.basicDark.deepDark}`,
            },
            hover: {
              background: theme.palette.basicDark.dark,
              border: "none",
            },
            active: {
              background: theme.palette.basicDark.dark,
              boxShadow: theme.shadow["3dShadowBasicActive"],
              border: "none",
            },
          };
        case "danger":
          return {
            normal: {
              background: theme.palette.danger.main,
              color: theme.palette.text.white,
              boxShadow: theme.shadow["3dShadowDanger"],
              border: `1px solid ${theme.palette.danger.dark}`,
            },
            hover: {
              background: theme.palette.danger.dark,
              border: "none",
            },
            active: {
              background: theme.palette.danger.dark,
              boxShadow: theme.shadow["3dShadowActive"],
              border: "none",
            },
          };
        case "clear":
          return {
            normal: {
              background: "none",
              color: theme.palette.black,
              boxShadow: "none",
              border: "none",
            },
            hover: {
              background: theme.palette.gray.light,
              border: "none",
            },
            active: {
              background: theme.palette.gray.main,
              boxShadow: "none",
              border: "none",
            },
          };
        case "primaryPale":
          return {
            normal: {
              background: theme.palette.primaryPale.main,
              color: theme.palette.primaryPale.dark,
              boxShadow: theme.shadow["3dShadowPrimaryPale"],
              border: `1px solid ${theme.palette.primaryPale.light}`,
            },
            hover: {
              background: theme.palette.primaryPale.highlight,
              border: "none",
            },
            active: {
              background: theme.palette.primaryPale.highlight,
              boxShadow: theme.shadow["3dShadowPrimaryPaleActive"],
              border: "none",
            },
          };
        default:
          return {
            normal: {
              background: theme.palette.primary.main,
              color: theme.palette.text.white,
              boxShadow: theme.shadow["3dShadowPrimary"],
              border: `1px solid ${theme.palette.primary.dark}`,
            },
            hover: {
              background: theme.palette.primary.dark,
              border: "none",
            },
            active: {
              background: theme.palette.primary.dark,
              boxShadow: theme.shadow["3dShadowActive"],
              border: "none",
            },
          };
      }
    };
    
    const colorStyle = getColorStyle(color);

    const anchorProps: {
      as?: string | React.ComponentType<{ className: string; children?: React.ReactNode }>;
      href?: string;
      target?: "_blank" | "_self" | "_parent" | "_top";
      rel?: string;
    } = {};

    if (typeof rest.href === "string") {
      const { href, target, rel } = rest;
      anchorProps.as = component || "a";
      anchorProps.href = href;
      anchorProps.target = target;
      anchorProps.rel = rel;
    }

    const buttonStyle = {
      display: inline ? "inline-flex" : "flex",
      justifyContent: "center",
      alignItems: "center",
      height: ButtonHeight[size],
      width: inline ? "fit-content" : "100%",
      paddingInline: paddingInline[size],
      borderRadius: BorderRadius[size],
      border: colorStyle.normal.border,
      background: colorStyle.normal.background,
      color: colorStyle.normal.color,
      textAlign: "center" as const,
      fontSize: FontSize[size],
      boxShadow: colorStyle.normal.boxShadow,
      transition: "background 0.3s",
      margin: 0,
      cursor: "pointer",
      outline: "none",
      textDecoration: "none",
      letterSpacing: "inherit",
      fontFamily: "inherit",
      lineHeight: "inherit",
    };

    if (typeof rest.href === "string") {
      const { href, target, rel } = rest;
      
      return (
        <a
          ref={ref as React.Ref<HTMLAnchorElement>}
          href={href}
          target={target}
          rel={rel}
          style={buttonStyle}
        >
          {icon && icon}
          {icon && children && <Spacer pr={0.5} />}
          {children}
        </a>
      );
    }
    
    return (
      <button
        ref={ref as React.Ref<HTMLButtonElement>}
        style={buttonStyle}
        {...rest}
      >
        {icon && icon}
        {icon && children && <Spacer pr={0.5} />}
        {children}
      </button>
    );
  }
);

export default ButtonWithRef;
