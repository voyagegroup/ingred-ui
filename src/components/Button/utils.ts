import { Theme } from "../../themes";
import { ButtonColor, ButtonColorStyle } from "./Button";

export const getContainerColorStyles = (
  theme: Theme,
): { [P in ButtonColor]: ButtonColorStyle } => ({
  primary: {
    normal: {
      background: theme.palette.primary.main,
      color: theme.palette.text.white,
      boxShadow: theme.shadow["3dShadowPrimary"],
      border: `1px solid ${theme.palette.primary.dark}`,
    },
    hover: {
      background: theme.palette.primary.dark,
      border: "none",
    },
    active: {
      background: theme.palette.primary.dark,
      boxShadow: theme.shadow["3dShadowActive"],
      border: "none",
    },
  },
  basicLight: {
    normal: {
      background: theme.palette.background.default,
      color: theme.palette.black,
      boxShadow: theme.shadow["3dShadowBasic"],
      border: `1px solid ${theme.palette.divider}`,
    },
    hover: {
      background: theme.palette.gray.highlight,
      border: `1px solid ${theme.palette.divider}`,
    },
    active: {
      background: theme.palette.gray.highlight,
      boxShadow: theme.shadow["3dShadowActive"],
      border: `1px solid ${theme.palette.divider}`,
    },
  },
  basicDark: {
    normal: {
      background: theme.palette.basicDark.main,
      color: theme.palette.black,
      boxShadow: theme.shadow["3dShadowBasic"],
      border: `1px solid ${theme.palette.basicDark.deepDark}`,
    },
    hover: {
      background: theme.palette.basicDark.dark,
      border: "none",
    },
    active: {
      background: theme.palette.basicDark.dark,
      boxShadow: theme.shadow["3dShadowBasicActive"],
      border: "none",
    },
  },
  danger: {
    normal: {
      background: theme.palette.danger.main,
      color: theme.palette.text.white,
      boxShadow: theme.shadow["3dShadowDanger"],
      border: `1px solid ${theme.palette.danger.dark}`,
    },
    hover: {
      background: theme.palette.danger.dark,
      border: "none",
    },
    active: {
      background: theme.palette.danger.dark,
      boxShadow: theme.shadow["3dShadowActive"],
      border: "none",
    },
  },
  clear: {
    normal: {
      background: "none",
      color: theme.palette.black,
      boxShadow: "none",
      border: "none",
    },
    hover: {
      background: theme.palette.gray.light,
      border: "none",
    },
    active: {
      background: theme.palette.gray.main,
      boxShadow: "none",
      border: "none",
    },
  },
  primaryPale: {
    normal: {
      background: theme.palette.primaryPale.main,
      color: theme.palette.primaryPale.dark,
      boxShadow: theme.shadow["3dShadowPrimaryPale"],
      border: `1px solid ${theme.palette.primaryPale.light}`,
    },
    hover: {
      background: theme.palette.primaryPale.highlight,
      border: "none",
    },
    active: {
      background: theme.palette.primaryPale.highlight,
      boxShadow: theme.shadow["3dShadowPrimaryPaleActive"],
      border: "none",
    },
  },
});
