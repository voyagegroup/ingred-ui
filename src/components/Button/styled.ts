import styled from "styled-components";

import { BaseButton } from "./internal/BaseButton";
import { ButtonColorStyle } from "./Button";

import { hexToRgba } from "../../utils/hexToRgba";

export type ContainerProps = ButtonColorStyle & {
  inline: boolean;
  fontSize: string;
  fontWeight: string;
  verticalPadding: string;
  horizontalPadding: string;
  paddingTopAtActive: string;
  paddingBottomAtActive: string;
  href?: string;
  disabled?: boolean;
};

export const ButtonContainer = styled(BaseButton)<ContainerProps>`
  display: ${({ inline }) => (inline ? "inline-flex" : "flex")};
  justify-content: center;
  align-items: center;
  padding: ${({ verticalPadding, horizontalPadding }) =>
    `${verticalPadding} ${horizontalPadding}`};
  width: ${({ inline }) => (inline ? "auto" : "100%")};
  border-radius: ${({ theme }) => theme.radius}px;
  border: ${({ normal, disabled, theme }) =>
    disabled ? `1px solid ${theme.palette.divider}` : normal.border};
  background: ${({ normal, disabled, theme }) =>
    disabled ? theme.palette.gray.light : normal.background};
  color: ${({ normal, disabled, theme }) =>
    disabled ? theme.palette.text.disabled : normal.color};
  text-align: center;
  font-weight: ${({ fontWeight }) => fontWeight};
  font-size: ${({ fontSize }) => fontSize};
  box-shadow: ${({ normal, disabled, theme }) =>
    disabled
      ? `0px -2px ${hexToRgba(
          theme.palette.black,
          0.16,
        )} inset, 0px 2px ${hexToRgba(theme.palette.black, 0.08)}`
      : normal.boxShadow};
  transition: background 0.3s;

  &:hover:not([disabled]) {
    background: ${({ hover }) => hover.background};
  }

  &:active:not([disabled]) {
    padding-top: ${({ paddingTopAtActive }) => paddingTopAtActive};
    padding-bottom: ${({ paddingBottomAtActive }) => paddingBottomAtActive};
    background: ${({ active }) => active.background};
    box-shadow: ${({ active }) => active.boxShadow};
  }
`;
