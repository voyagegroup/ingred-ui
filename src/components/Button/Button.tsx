import * as React from "react";
import { fontSize } from "../Typography/Typography";

import { Props as BaseButtonProps } from "./internal/BaseButton";
import * as Styled from "./styled";
import { Theme, useTheme } from "../../themes";
import { hexToRgba } from "../../utils/hexToRgba";

export type ButtonSize = "small" | "medium" | "large";
export type ButtonColor = "primary" | "secondary" | "danger" | "clear";

export type ButtonColorStyle = {
  normal: {
    background: string;
    color: string;
    boxShadow: string;
    border: string;
  };
  hover: {
    background?: string;
    border: string;
  };
  active: {
    background?: string;
    boxShadow: string;
    border: string;
  };
};

export type Padding = {
  theme: Theme;
  size: ButtonSize;
  color: ButtonColor;
};

const getContainerColorStyles = (
  theme: Theme,
): { [P in ButtonColor]: ButtonColorStyle } => ({
  primary: {
    normal: {
      background: theme.palette.primary.main,
      color: theme.palette.text.white,
      boxShadow: `0px -2px ${hexToRgba(
        theme.palette.black,
        0.16,
      )} inset, 0px 2px ${hexToRgba(theme.palette.black, 0.08)}`,
      border: `1px solid ${theme.palette.primary.dark}`,
    },
    hover: {
      background: theme.palette.primary.dark,
      border: "none",
    },
    active: {
      background: theme.palette.primary.dark,
      boxShadow: `inset 0 2px ${hexToRgba(theme.palette.black, 0.16)}`,
      border: "none",
    },
  },
  secondary: {
    normal: {
      background: theme.palette.background.default,
      color: theme.palette.black,
      boxShadow: `0px -2px ${hexToRgba(
        theme.palette.black,
        0.16,
      )} inset, 0px 2px ${hexToRgba(theme.palette.black, 0.08)}`,
      border: `1px solid ${theme.palette.divider}`,
    },
    hover: {
      background: theme.palette.gray.highlight,
      border: `1px solid ${theme.palette.divider}`,
    },
    active: {
      background: theme.palette.gray.highlight,
      boxShadow: `inset 0 2px ${hexToRgba(theme.palette.black, 0.16)}`,
      border: `1px solid ${theme.palette.divider}`,
    },
  },
  danger: {
    normal: {
      background: theme.palette.danger.main,
      color: theme.palette.text.white,
      boxShadow: `0px -2px ${hexToRgba(
        theme.palette.black,
        0.16,
      )} inset, 0px 2px ${hexToRgba(theme.palette.black, 0.08)}`,
      border: `1px solid ${theme.palette.danger.dark}`,
    },
    hover: {
      background: theme.palette.danger.dark,
      border: "none",
    },
    active: {
      background: theme.palette.danger.dark,
      boxShadow: `inset 0 2px ${hexToRgba(theme.palette.black, 0.16)}`,
      border: "none",
    },
  },
  clear: {
    normal: {
      background: "none",
      color: theme.palette.gray.deepDark,
      boxShadow: "none",
      border: "none",
    },
    hover: {
      background: theme.palette.gray.light,
      border: "none",
    },
    active: {
      background: theme.palette.gray.main,
      boxShadow: "none",
      border: "none",
    },
  },
});

const paddingAtActive: Record<
  ButtonSize,
  { paddingTop: string; paddingBottom: string }
> = {
  small: {
    paddingTop: "7px",
    paddingBottom: "5px",
  },
  medium: {
    paddingTop: "11px",
    paddingBottom: "9px",
  },
  large: {
    paddingTop: "14px",
    paddingBottom: "12px",
  },
};

const verticalPadding: Record<ButtonSize, { padding: string }> = {
  small: {
    padding: "6px",
  },
  medium: {
    padding: "10px",
  },
  large: {
    padding: "13px",
  },
};

const getPadding = ({ theme, size, color }: Padding) => ({
  horizontalPadding: size === "small" ? `10px` : `${theme.spacing * 2}px`,
  verticalPadding: verticalPadding[size].padding,
  paddingTopAtActive: color === "clear" ? "" : paddingAtActive[size].paddingTop,
  paddingBottomAtActive:
    color === "clear" ? "" : paddingAtActive[size].paddingBottom,
});

export type ButtonProps = Omit<BaseButtonProps, "color"> & {
  /**
   * The component used for the root node.
   * Default: `<button />`
   */
  component?:
    | keyof JSX.IntrinsicElements
    | React.ComponentType<{ className: string }>;
  color?: ButtonColor;
  /**
   * Control whether "inline" or "block" Element.
   */
  inline?: boolean;
  size?: ButtonSize;
  onClick?: (event: React.MouseEvent<Element, MouseEvent>) => void;
  /**
   * If added this props, root node becomes `<a />`.
   */
  href?: string;
};

const Button = React.forwardRef<HTMLElement, ButtonProps>(
  (
    {
      component,
      children,
      color = "primary",
      inline = false,
      size = "medium",
      href,
      ...rest
    },
    ref,
  ) => {
    const theme = useTheme();
    const colorStyle = getContainerColorStyles(theme)[color];
    const {
      horizontalPadding,
      verticalPadding,
      paddingTopAtActive,
      paddingBottomAtActive,
    } = getPadding({
      theme,
      size,
      color,
    });

    const isLink = !!href;
    let anchorProps: any = {};
    if (isLink) {
      anchorProps = {
        as: component || "a",
        href,
      };
    }

    return (
      <Styled.ButtonContainer
        ref={ref}
        {...rest}
        as={component || "button"}
        {...anchorProps}
        inline={inline}
        verticalPadding={verticalPadding}
        horizontalPadding={horizontalPadding}
        paddingTopAtActive={paddingTopAtActive}
        paddingBottomAtActive={paddingBottomAtActive}
        normal={{ ...colorStyle.normal }}
        hover={{ ...colorStyle.hover }}
        active={{ ...colorStyle.active }}
        fontWeight={color === "secondary" ? "normal" : "bold"}
        fontSize={
          size === "small" ? `${fontSize["xs"]}px` : `${fontSize["md"]}px`
        }
      >
        {children}
      </Styled.ButtonContainer>
    );
  },
);

export default Button;
