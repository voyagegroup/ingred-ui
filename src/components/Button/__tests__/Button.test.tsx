import * as React from "react";
import "@testing-library/jest-dom";
import { cleanup } from "@testing-library/react";
import Button from "..";
import { renderWithThemeProvider } from "../../../utils/renderWithThemeProvider";

describe("Button component testing", () => {
  afterEach(cleanup);

  test("Button", () => {
    const { asFragment } = renderWithThemeProvider(<Button />);
    expect(asFragment()).toMatchSnapshot();
  });
});
