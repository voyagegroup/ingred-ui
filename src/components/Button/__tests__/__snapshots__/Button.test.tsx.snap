// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component testing Button 1`] = `
<DocumentFragment>
  <button
    class="sc-gsDKAQ sc-dkPtRN iVvbbF klzIpQ"
    font-size="14px"
    font-weight="bold"
  />
</DocumentFragment>
`;
